src/Components/Browser.JS/src/Rendering/Renderer.ts

```diff
@@ -1,13 +1,18 @@
+/* eslint-disable @typescript-eslint/camelcase */
 import { System_Object, System_String, System_Array, MethodHandle, Pointer } from '../Platform/Platform';
 import { platform } from '../Environment';
 import { RenderBatch } from './RenderBatch/RenderBatch';
-import { BrowserRenderer } from './BrowserRenderer';
+import { BrowserRenderer, StartEndPair } from './BrowserRenderer';
 
-type BrowserRendererRegistry = { [browserRendererId: number]: BrowserRenderer };
+interface BrowserRendererRegistry {
+  [browserRendererId: number]: BrowserRenderer;
+}
 const browserRenderers: BrowserRendererRegistry = {};
 
-export function attachRootComponentToElement(browserRendererId: number, elementSelector: string, componentId: number) {
-  const element = document.querySelector(elementSelector);
+export function attachRootComponentToElement(browserRendererId: number, elementSelector: string | StartEndPair, componentId: number): void {
```

Comment:
This no longer seems to be the desired API to match our use cases. This can get called in two ways:

 * From JS interop, passing a selector string
 * From code in CircuitRegistry.ts, passing a StartEndPair

It would be much easier to understand if these were two different methods, so the code doesn't have to do runtime inspection to infer what kind of parameter is being passed, considering that we can just change the one in `CircuitRegistry` to call the variant that takes a `StartEndPair`.

How about:

 * attachRootComponentToElement (takes an element selector string)
 * attachRootComponentToLogicalElement (takes a logical element, or change this to "takes a start end pair" if that's necessary, though I'm still trying to understand why that's needed instead of a logical element)

The first of those would work just by converting the running the selector and obtaining a logical element, then calling the second.

Reply:
Ahh, I just read this suggestion. Makes sense. Let's do this.